src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -129,7 +129,17 @@ public SubmissionDto submitSolution(String roomId, SubmissionRequest request) {
             throw new ApiException(GameError.INVALID_PERMISSIONS);
         }
 
-        return submitService.submitSolution(game, request);
+        SubmissionDto submissionDto = submitService.submitSolution(game, request);
+
+        /**
+         * Check allSolved if the user solved the problem and not everybody
+         * has solved the problem yet.
+         */
+        if (submissionDto != null && submissionDto.getNumCorrect().equals(submissionDto.getNumTestCases())&& Boolean.FALSE.equals(game.getAllSolved())) {
+            conditionalSolvedSocketMessage(game);
+        }
+        
```

Comment:
Thoughts on removing this if statement check? Since the method handles the conditional update step as well (so this check might just be saving a few ms of time)

Reply:
The method `conditionalSolvedSocketMessage` doesn't actually check for the conditions that `submissionDto` exists, the player got full credit, and not everybody has already solved the game -- these are the conditions I'm checking for here. I could move those checks within the `conditionalSolvedSocketMessage` method though -- should I do that?

Reply:
The main reason I brought it up was cause 1. the submission shouldn't ever return null based on our implementation 2. `conditionalSolvedSocketMessage` already loops through each player to check if each `player.solved` is true (tho I can see why that could be there if we wanted to save a bit on runtime to prevent running the check on all submissions), and 3. seems like the game error not found thing you brought up in a different comment

But yeah if you decide to keep it I think it'd be best to leave here that way you don't have to pass in the submission to that method

Reply:
I can address both of these:
1. Yeah this is true, this is actually used for kind of a hacky purpose: in the test of this method in `GameManagementServiceTests.java`, the `submissionDto` actually is `null` if you don't mock it, so I used this check to prevent an error from occurring. I'm happy to instead mock the `submissionDto` return, just let me know.
2. That's a good point, because that only changes
3. This is fair, though this was my response to that comment:
> Not in our current setup -- the situation in which it would be useful is if the game did not end when everybody solves the problem, but rather when the time runs out if players are competing to make the best solution (fastest or least memory). Do you think I should keep this in for that reason, or no?

Reply:
Continued in the other chain

Reply:
I've addressed (1) as I've removed the `submissionDto != null` check, so now only the second two are applicable.